common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -101,9 +114,31 @@ class HTTPClient(
             connection.disconnect()
         }
 
-        val responseBody = payload?.let { JSONObject(it) } ?: throw IOException("Network call payload is null.")
         log(LogIntent.DEBUG, NetworkStrings.API_REQUEST_COMPLETED.format(connection.requestMethod, path, responseCode))
-        return Result(responseCode, responseBody)
+        if (payload == null) {
+            throw IOException("Network call payload is null.")
```

Comment:
I know this was here before, but maybe this should be "http response payload is null"? so that it's clear that the empty payload is from the response, not the request

Reply:
also, should the string be in a strings file?